python/taichi/lang/matrix.py

```diff
@@ -256,7 +292,11 @@ def linearize_entry_id(self, *args):
     def __call__(self, *args, **kwargs):
         _taichi_skip_traceback = 1
         assert kwargs == {}
-        return self.entries[self.linearize_entry_id(*args)]
+        if self.local_tensor_proxy is not None:
```

Comment:
I believe this branch is redundant - this path is already prohibited in `subscript()`.

Reply:
https://github.com/taichi-dev/taichi/pull/2637/commits/0262df20e03d21b21d9b8f87d7c0ea6552578819